src/geocel/GeantGeoParams.cc

```diff
@@ -309,70 +246,96 @@ std::vector<inp::Volume> make_inp_volumes(GeantGeoParams const& geo)
     result.resize(vol_labels.size());
 
     // Process each logical volume
-    for (auto vol_id : range(ImplVolumeId{vol_labels.size()}))
+    for (auto iv_id : range(ImplVolumeId{vol_labels.size()}))
     {
-        auto const& label = vol_labels.at(vol_id);
+        auto const& label = vol_labels.at(iv_id);
         if (label.empty())
         {
             // This volume isn't part of the world hierarchy
             continue;
         }
 
-        auto* g4lv = geo.id_to_geant(vol_id);
-        CELER_ASSERT(g4lv);
-        result[vol_id.get()] = inp_from_geant(geo, label, *g4lv);
+        auto vol_id = geo.volume_id(iv_id);
+        auto& g4lv = *geo.id_to_geant(vol_id);
+
+        // TODO: do we want to strip pointers?
+        auto& vol_inp = result[vol_id.get()];
+        vol_inp.label = label;
+        vol_inp.material = [&geo, mat = g4lv.GetMaterial()]() -> GeoMatId {
+            if (!mat)
+                return {};
+            return geo.geant_to_id(*mat);
+        }();
+
+        // Populate volume.children with child volume instance IDs
+        auto num_children = g4lv.GetNoDaughters();
+        vol_inp.children.reserve(num_children);
+        for (auto i : range(num_children))
+        {
+            // One physical volume can correspond to multiple volume instances
+            // if using replica/parameterized volumes
+            G4VPhysicalVolume* g4pv = g4lv.GetDaughter(i);
+            CELER_ASSERT(g4pv);
+            for (int j = 0, jmax = g4pv->GetMultiplicity(); j < jmax; ++j)
+            {
+                if (g4pv->IsReplicated())
+                {
+                    g4pv->SetCopyNo(j);
+                }
+                auto vol_inst_id = geo.geant_to_id(*g4pv);
+                vol_inp.children.push_back(vol_inst_id);
+            }
```

Comment:
Is the intent to have at the end of the loop `g4pv`' copy number to `jmax-1` or should it be reset or does it not matter?